contracts/DEX.sol

```diff
@@ -24,13 +26,19 @@ contract DEX {
         uint256 ssvAmount = amount / rate;
         cdtToken.transferFrom(msg.sender, address(this), amount);
```

Comment:
Instead of transferring `amount`, we should transfer `ssvAmount * rate`.
This way we're not locking remainders in the contract.
e.g.
rate = 10
user sends 15 CDT tokens, gets one SSV token in exchange but still sends 15.

Reply:
thank you, fixed